coffee-backend/src/main/kotlin/chat/letscoffee/security/security/UserPrincipal.kt

```diff
@@ -0,0 +1,70 @@
+package chat.letscoffee.security.security
+
+import chat.letscoffee.security.model.User
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.oauth2.core.user.OAuth2User
+
+class UserPrincipal(val id: Long?, val email: String, private val password: String?, private val authorities: Collection<GrantedAuthority>) : OAuth2User, UserDetails {
```

Comment:
I'm not so sure how smart it is to have both of these entities as the same class. They serve a bit different purposes; the principal is for authentication while the User is for transporting data about the user in the application. Let's leave it as it is for now, but keep in mind that we might want to split these classes at a later point.

Reply:
I agree